pandas/sparse/frame.py

```diff
@@ -159,7 +159,7 @@ def _init_dict(self, data, index, columns, dtype=None):
                     v = [v.get(i, nan) for i in index]
 
                 v = sp_maker(v)
-            sdict[k] = v
+            sdict[k] = v.astype(np.dtype(dtype)) if dtype is not None else v
```

Comment:
you would do this in ``sp_maker``, also don't use if/else on a single line (hard to read)